### PR TITLE
Downtime script for pyops

### PR DESCRIPTION
**Pull request by khappone**
Added downtime script for pyops. Cleaned up pyops to be nicer python.

Fixed indetation

### CODE REVIEW

File: pyops.py [hunk 1]
```diff
@@ -7,381 +7,410 @@
...
+        '''
+        result = self.post_data(json.loads('{ "starttime" : "%s", "endtime" : "%s", "comment" : "%s" }' % (starttime, endtime, comment)),"downtime?hostgroupid=%s" % hg_id)
+        self.log.debug(`"downtime : downtime ", result`)
+        self.log.info(result)
+
+    def set_host_downtime(self, hosts, starttime, endtime, comment):
+        '''
+        Create downtime for the hostlist.
+
+        Lessons learned: Cloud NOT set downtime by hostid.
```
Comment (Aegil) [inline pyops.py]: Could Cloud

**Comment by khappone** [comment]
Btw, this now has our monitor server name in github. I thought it would be ok? but I can remove it. It makes the tool a bit more cumbersome to use though.

File: set_downtime.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+# Then in bash run
+# $ export prod_ops_user=opsview_username
+# $ read -s prod_ops_pass
+# $ export prod_ops_pass
+
+def main(argv=None):
+    if argv is None:
+        argv = sys.argv
+
+    parser = argparse.ArgumentParser(description='Set downtime for hosts in opeview, starting now.')
```
Comment (martbhell) [inline set_downtime.py]: Tiny Typo opeview.

File: set_downtime.py [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    parser.add_argument('-c', '--comment', dest="comment", default="Dowtime set by pyops", help="Comment for the downtime")
+    parser.add_argument('hosts', nargs='*', help="List of hosts to apply the downtime to.")
```
Comment (martbhell) [inline set_downtime.py]: I guess this list have to be space separated. With commas I got:

<pre>
$ python set_downtime.py somenode1,somenode2 -c "mycomment" -e "2018-07-17 09:58:44"
No handlers could be found for logger "pyops"
</pre>

Not sure if that's a bug that one can fix easily.

Using spaces it worked anyway :)

Are there any other formats it accepts a list of hosts in?

**Comment by khappone** [comment]
Yes, good comments. Made the help a bit better. Also fixed logging so it'll write any problems to stderr. In the same go, I added a parameter so you can specify the starttime too.

**Pull request merged**